### ISSUE TITLE
Can't sign in to settings sync with GitHub if using non-default profile and `--user-data-dir`

### ISSUE THREAD

**Issue by o-l-a-v** [microsoft/vscode#209815]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.88.0 (system setup) x64
- OS Version: Windows 11 23H2

Steps to Reproduce:

1. Create a new profile "Personal" and user data dir "%USERPROFILE%/.vscode-personal"
2. Start VSCode from CMD with `code --user-data-dir "%userprofile%\.vscode-personal" --profile "Personal"`
3. Try to sign in to settings sync with GitHub account.
   * Web browser opens.
   * You sign in.
   * After successfully signing in a new VSCode windows using default profile and default data path opens.
4. Result: You can't sign in to settings sync with GitHub if using non-default profile with non-default data path.

Observation: Settings sync with Microsoft account works though.

**Comment by o-l-a-v** [comment]
Using the backup sign-in method of device login works.

Either:

* Make sure the preferred method supports non-default `--user-data-dir`, and redirects back to correct VSCode instance, OR:
* Detect non default `--user-data-dir` and use device auth.

**Comment by TylerLeonhardt** [comment]
Yeah, this is because of `vscode://` handling... a different user data dir is really a different instance of VS Code.

Really our URL handling should probably be smart enough to route to the right one... but I guess we could check if the `--user-data-dir` flag was used and if so, just use device code flow.

The flow code is here:
https://github.com/microsoft/vscode/blob/6184b4590c5a10fae0c410d48df2cf66cdefcc42/extensions/github-authentication/src/flows.ts#L24

If you're interested in contributing.
